src/Hit/Git.hs

```diff
@@ -191,3 +196,23 @@ issueFromBranch =
     . T.takeWhile isDigit
     . T.drop 1
     . T.dropWhile (/= '/')
+
+{- | Perform given action by adding all untracked files to index and returning
+them back after action.
+-}
+withUntrackedFiles :: IO a -> IO a
+withUntrackedFiles action = bracket
+    addUntrackedFiles
+    removeUntrackedFiles
+    (const  action)
+  where
+    -- Add all untracked file to index so they will appear in diff
+    addUntrackedFiles :: IO [Text]
+    addUntrackedFiles = do
+        untrackedFiles <- lines <$> "git" $| ["ls-files", "--others", "--exclude-standard"]
+        for_ untrackedFiles $ \file -> void $ "git" $| ["add", file]
+        pure untrackedFiles
+
+    -- Return untracked files back to not spoil git state and have unexpected behavior
+    removeUntrackedFiles :: [Text] -> IO ()
+    removeUntrackedFiles = mapM_ $ \file -> void $ "git" $| ["reset", file]
```

Comment:
This won't affect files that were tracked before, will it?

Reply:
@vrom911 Not, it doesn't affect files that are already tracked. List of `[Text]` here is the files that weren't tracked and returned by the `addUntrackedFiles` function. This function works only with untracked files.